pkg/tar/tar.go

```diff
@@ -38,10 +38,10 @@ var ErrNotSupportedPlatform = errors.New("platform and architecture is not suppo
 // root of the tar file and should be cleaned (for example using filepath.Clean)
 type PathWhitelistMap map[string]struct{}
 
-// extractTar extracts a tarball (from a tar.Reader) into the root directory
-// if pwl is not nil, only the paths in the map are extracted.
-// If overwrite is true, existing files will be overwritten.
-func extractTar(tr *tar.Reader, overwrite bool, pwl PathWhitelistMap, uidRange *uid.UidRange) error {
+// ExtractTarInsecure extracts a tarball (from a tar.Reader) into the root
+// directory if pwl is not nil, only the paths in the map are extracted. If
+// overwrite is true, existing files will be overwritten.
+func ExtractTarInsecure(tr *tar.Reader, overwrite bool, pwl PathWhitelistMap, uidRange *uid.UidRange) error {
```

Comment:
Sorry, thickening the plot once again.

`pkg/tar/chroot.go` moves somewhere like `common/tar/chroot.go`
Then in acbuild you implement a version of that file, but it's simpler - no need for uid range or for the overwrite flag (I don't think?)
`func ExtractTar(rs io.Reader, dir string, pwl PathWhitelistMap) error`

Sorry for the yak shave but this is cleanup that I think has been needed for a long time. Packages under pkg/ should not have anything rkt specific and this one feels a bit leaky.